specutils/spectra/spectrum_mixin.py

```diff
@@ -208,14 +215,27 @@ def with_spectral_unit(self, unit, velocity_convention=None,
                      even if your spectrum has air wavelength units
 
         """
-        new_wcs, new_meta = self._new_spectral_wcs(
-            unit=unit,
-            velocity_convention=velocity_convention or self.velocity_convention,
-            rest_value=rest_value or self.rest_value)
 
-        spectrum = self.__class__(flux=self.flux, wcs=new_wcs, meta=new_meta)
+        velocity_convention = velocity_convention if velocity_convention is not None else self.velocity_convention # noqa
+        rest_value = rest_value if rest_value is not None else self.rest_value
+        unit = self._new_wcs_argument_validation(unit, velocity_convention,
+                                                 rest_value)
+
+        # Store the original unit information and WCS for posterity
+        meta = self._meta.copy()
+
+        if 'original_unit' not in self._meta:
+            orig_unit = self.wcs.unit[0] if hasattr(self.wcs, 'unit') else self.spectral_axis.unit
+            meta['original_unit'] = orig_unit
+
+        if 'original_wcs' not in self.meta:
+            meta['original_wcs'] = self.wcs.copy()
```

Comment:
```suggestion
            meta['original_wcs'] = self.wcs.deepcopy()
```
Since this brought up the issue of the convenience copy method, you might actually want this, if you really treat the  created spectrum as a new object; otherwise any changes to `new_spec.wcs` will modify the original as well.
Should perhaps be added to the test below for an instance with wcs.

Reply:
Good call, I can never keep straight when copy vs deepcopy is the right answer.